### PR TITLE
draws bus routes

### PR DESCRIPTION
**Pull request by mlelien**
forgot to pull from master

### CODE REVIEW

File: big-red-shuttle/Network.swift [hunk 1]
```diff
@@ -174,6 +174,40 @@ public func getStops() ->  [Stop] {
...
+    func getPolyline(waypoints:[Stop], origin:Stop, end:Stop) {
+        let baseURLDirections = "https://maps.googleapis.com/maps/api/directions/json?"
+        let origin = "\(origin.lat),\(origin.long)"
+        let destination = "\(end.lat),\(end.long)"
+        
+        var directionsURLString = baseURLDirections + "origin=" + origin + "&destination=" + destination
+        
+        if waypoints.count > 0 {
+            directionsURLString += "&waypoints=optimize:true"
+            for i in 0...waypoints.count - 1 {
```
Comment (anniexcheng) [inline big-red-shuttle/Network.swift]: `0...<waypoints.count`

File: big-red-shuttle/Network.swift [hunk 2]
```diff
@@ -174,6 +174,40 @@ public func getStops() ->  [Stop] {
...
+                let coords = "\(waypoints[i].lat),\(waypoints[i].long)"
+                directionsURLString +=  "|" + coords
+            }
+        }
+        
+        directionsURLString = directionsURLString.addingPercentEncoding(withAllowedCharacters: NSCharacterSet.urlQueryAllowed)!
```
Comment (anniexcheng) [inline big-red-shuttle/Network.swift]: don't need `NSCharacterSet`

File: big-red-shuttle/Network.swift [hunk 3]
```diff
@@ -174,6 +174,40 @@ public func getStops() ->  [Stop] {
...
+        
+        if let directionsURL = URL(string: directionsURLString) {
+            if let data = try? Data(contentsOf: directionsURL) {
+                let json = JSON(data: data)
+                let status = json["status"].stringValue
+                if status == "OK" {
+                    self.overviewPolyline = json["routes"][0]["overview_polyline"]["points"].stringValue
```
Comment (anniexcheng) [inline big-red-shuttle/Network.swift]: no `self`

File: big-red-shuttle/Network.swift [hunk 4]
```diff
@@ -174,6 +174,40 @@ public func getStops() ->  [Stop] {
...
+            }
+        }
+        
+        directionsURLString = directionsURLString.addingPercentEncoding(withAllowedCharacters: NSCharacterSet.urlQueryAllowed)!
+        
+        if let directionsURL = URL(string: directionsURLString) {
+            if let data = try? Data(contentsOf: directionsURL) {
+                let json = JSON(data: data)
+                let status = json["status"].stringValue
+                if status == "OK" {
```
Comment (anniexcheng) [inline big-red-shuttle/Network.swift]: You might want to make a `Status` enum, and add in some error handling if status is not "OK"

File: big-red-shuttle/StopsViewController.swift [hunk 5]
```diff
@@ -61,7 +64,33 @@ class StopsViewController: UIViewController, GMSMapViewDelegate {
             marker.map = mapView
         }
     }
+    
+    
+    //drawing
+    func initPolyline() {
+        let stops = getStops()
+        let stopsA = Array(stops[1..<9]) //can only have a max of 8 waypoints (i.e. stops that don't include start and end)
```
Comment (anniexcheng) [inline big-red-shuttle/StopsViewController.swift]: Will this always be equal to the number of stops? If so, we should base this range on the length of the array where we store the different stops

File: big-red-shuttle/StopsViewController.swift [hunk 6]
```diff
@@ -61,7 +64,33 @@ class StopsViewController: UIViewController, GMSMapViewDelegate {
...
+        
+        self.polyline.getPolyline(waypoints: stopsA, origin:stops[0], end:stops[9])
+        self.polyline2.getPolyline(waypoints: [], origin: stops[9], end: stops[0])
+        self.drawRoute()
```
Comment (anniexcheng) [inline big-red-shuttle/StopsViewController.swift]: lines 74-76, no `self`

File: big-red-shuttle/StopsViewController.swift [hunk 7]
```diff
@@ -16,6 +16,8 @@ class StopsViewController: UIViewController, GMSMapViewDelegate {
     var mapView: GMSMapView!
     var panBounds: GMSCoordinateBounds!
     let kBoundPadding: CGFloat = 40
+    var polyline = Polyline()
+    var polyline2 = Polyline()
```
Comment (anniexcheng) [inline big-red-shuttle/StopsViewController.swift]: Are `polyline` and `polyline2` ever altered? If not, use `let`.

File: big-red-shuttle/StopsViewController.swift [hunk 8]
```diff
@@ -61,7 +64,33 @@ class StopsViewController: UIViewController, GMSMapViewDelegate {
...
+    }
 
+    func drawRoute() {
+        let path1 = GMSMutablePath(fromEncodedPath: polyline.overviewPolyline)
+        let path2 = GMSMutablePath(fromEncodedPath: polyline2.overviewPolyline)
+        let routePolyline = GMSPolyline(path: path1)
+        let routePolyline2 = GMSPolyline(path: path2)
+        
+        routePolyline.strokeColor = UIColor(red: 206/255, green: 73/255, blue: 55/255, alpha: 1.0)
+        routePolyline2.strokeColor = UIColor(red: 206/255, green: 73/255, blue: 55/255, alpha: 1.0)
```
Comment (anniexcheng) [inline big-red-shuttle/StopsViewController.swift]: These colors will be added into the UIColor+Shared.swift file when Monica merges her stuff

File: big-red-shuttle/StopsViewController.swift (hunk 5 above)
Comment (mlelien) [reply big-red-shuttle/StopsViewController.swift]: Google maps only allows eight waypoints (stops that aren't origin or destination), so stopsA has a max of eight

Comment (anniexcheng) [reply big-red-shuttle/StopsViewController.swift]: ok in that case, make `9` a constant like `maxWaypoints` and use that constant instead

**Comment by anniexcheng** [comment]
Good to merge in!

**Comment by anniexcheng** [comment]
Merged in 9b532f3

**Pull request merged**